packages/components/src/ui/radix-dropdown/index.tsx

```diff
@@ -0,0 +1,177 @@
+/**
+ * External dependencies
+ */
+import * as DropdownMenuPrimitive from '@radix-ui/react-dropdown-menu';
+
+/**
+ * WordPress dependencies
+ */
+import { forwardRef } from '@wordpress/element';
+import { SVG, Circle } from '@wordpress/primitives';
+import { check, lineSolid } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import Icon from '../../icon';
+import * as DropdownMenuStyled from './styles';
+import type {
+	DropdownMenuProps,
+	DropdownSubMenuProps,
+	DropdownItemProps,
+} from './types';
+
+// Observations / Questions:
+// - is it enough on the larger components to have only one forwarded ref?
+//   If we have only one, should it be to the "root", the "trigger", or the "content"?
+// - Should we be consistent in using the same value for the `asChild` prop on:
+//   - trigger
+//   - content
+//   - sub trigger
+//   - sub content
+// - Should we allow customizing the `asChild` prop on trigger and content?
+// - Props & customisability:
+//   - Which props should we expose?
+//   - Should props be "namespaced" for each subcomponent?
+//   - We should probably explicitly `Pick<>` every prop that we want to expose
+// - Subtrigger arrow:
+//   - Should it always be there (ie. an internal implementation)?
+//   - Should we just expect that the the consumers handle it themselves?
+//   - Should we expose it as a separate component that consumers could use?
+// - DropdowmMenuItem icon prop:
+//   - added to mirror previous menu item component
+//   - should we expect consumers to provide this directly with children?
+//   - should we expose prefix / suffix to help?
```

Comment:
Prefix and overall alignment between menu items:
- should menu items have some left spacing to ensure that they will align nicely with items that have a prefix and checkbox/radio items?
- should checkbox/radio items allow for a prefix, despite the fact that they already show an icon to highlight their checked status?

Reply:
@mirka to explore

Reply:
> should menu items have some left spacing to ensure that they will align nicely with items that have a prefix and checkbox/radio items?

As of now, this is not implemented

> should checkbox/radio items allow for a prefix, despite the fact that they already show an icon to highlight their checked status?

As of now, I've deliberately removed `prefix` from checkbox and radio items

Reply:
Doesn't have to be in this initial PR, but I think I would like a context-based system to automatically align the item labels whenever there is a radio/checkbox item in the mix.

My rough idea of how that would be implemented is:

- The context would contain the state value `textIndentWidth` and state setter `setTextIndentWidth` (names TBD).
- The context would be shared within `DropdownMenu.Content`.
- `DropdownMenu.SubContent` should override the context with its own state and state setter.
- Radio and checkbox items would call `setTextIndentWidth()` with an appropriate value.
- `DropdownMenuItem` and `DropdownSubMenu` should align its text label accordingly from the context value (could be done with either `grid` column or `prefix` width).

Reply:
Marking as solved, although I left a link to this convo in the [tracking issue](https://github.com/WordPress/gutenberg/issues/50459)